src/scenic/syntax/scenic.gram

```diff
@@ -737,6 +741,71 @@ scenic_class_property_attribute: &&(
     | "dynamic" { s.Dynamic(LOCATIONS) }
 )
 
+
+# Behavior
+# --------
+
+scenic_behavior_def:
+    | "behavior" a=NAME b=['(' z=[params] ')' { z }] &&':' c=scenic_behavior_def_block {
+        s.BehaviorDef(
+            a.string,
+            args=b or self.make_arguments(None, [], None, [], None),
+            docstring=c[0],
+            header=c[1],
+            body=c[2],
+            LOCATIONS,
+        )
+     }
+
+scenic_behavior_def_block:
+    # if only preconditions and/or invariants are specified
+    | NEWLINE INDENT a=[x=STRING NEWLINE { x.string }] b=scenic_behavior_header DEDENT { (a, b, [ast.Pass()]) }
```

Comment:
This rule is needed to parse behaviors only with preconditions or invariants and without any plain statements.

e.g.

```
behavior B:
    precondition: condition
```

Reply:
Let's simplify the grammar by disallowing these: such behaviors already trigger an error at simulation time since they take no actions.

Reply:
Fixed in 9e4d682fdffffdc802a9aa7cc3d981d17a23ae6d